sklearn/preprocessing/tests/test_data.py

```diff
@@ -191,19 +210,269 @@ def test_scaler_2d_arrays():
+    # for large datasets with values of large magniture
+    rng = np.random.RandomState(0)
+    n_features = 2
+    n_samples = 100
+    offsets = rng.uniform(-1e15, 1e15, size=n_features)
+    scales = rng.uniform(1e3, 1e6, size=n_features)
+
+    chunk_size = 1
+    data_chunks = [rng.randn(chunk_size, n_features) * scales + offsets
+                   for i in range(n_samples)]
+
+    X = np.vstack(data_chunks)
+    scaler_batch = StandardScaler().fit(X)
+
+    scaler_incr = StandardScaler()
+    for i, data_chunk in enumerate(data_chunks):
+        scaler_incr = scaler_incr.partial_fit(data_chunk)
+
+    # Regardless of abs values, they must not be more diff 6 significant digits
+    tol = 10 ** (-6)
+    assert_allclose(scaler_incr.mean_, scaler_batch.mean_, rtol=tol)
+    assert_allclose(scaler_incr.var_, scaler_batch.var_, rtol=tol)
+    assert_allclose(scaler_incr.scale_, scaler_batch.scale_, rtol=tol)
+    # NOTE Be aware that for much larger offsets std is very unstable (last
+    # assert) while mean is OK.
+
+    # Sparse input
+    n_samples = 100
+    n_features = 3
+    scale = 1e20
+    data_chunks = [rng.random_integers(0, 1, n_features).astype(np.float64) *
+                   scale for i in range(n_samples)]
+
+    X = np.vstack(data_chunks)
+    X_csr = sparse.csr_matrix(X)
+    X_csc = sparse.csc_matrix(X)
+    # with_mean=False is required with sparse input
+    scaler_batch_csr = StandardScaler(with_mean=False).fit(X_csr)
+    scaler_batch_csc = StandardScaler(with_mean=False).fit(X_csc)
+
+    scaler_incr_csr = StandardScaler(with_mean=False)
+    scaler_incr_csc = StandardScaler(with_mean=False)
+    for data_chunk in data_chunks:
+        csr_chunk = sparse.csr_matrix(data_chunks)
+        csc_chunk = sparse.csc_matrix(data_chunks)
+        scaler_incr_csr = scaler_incr_csr.partial_fit(csr_chunk)
+        scaler_incr_csc = scaler_incr_csc.partial_fit(csc_chunk)
+
+    # Regardless of abs values, they must not be more diff 6 significant digits
+    tol = 10 ** (-6)
+    assert_true(scaler_batch_csr.mean_ is not None)
+    assert_true(scaler_batch_csc.mean_ is not None)
+    assert_allclose(scaler_incr_csr.var_, scaler_batch_csr.var_, rtol=tol)
+    assert_allclose(scaler_incr_csc.var_, scaler_batch_csc.var_, rtol=tol)
+    assert_allclose(scaler_incr_csr.scale_, scaler_batch_csc.scale_, rtol=tol)
+    assert_allclose(scaler_incr_csr.scale_, scaler_batch_csc.scale_, rtol=tol)
+
+
+def test_partial_fit_sparse_input():
+    # Check that sparsity is not destroyed
+    X = np.array([[1], [0], [0], [5]])
+    X_csr = sparse.csr_matrix(X)
+    X_csc = sparse.csc_matrix(X)
+
+    assert_array_almost_equal(X_csr.todense(), X_csr.todense())
+    assert_array_almost_equal(X_csc.todense(), X_csc.todense())
+    assert_true(sparse.issparse(X_csr))
+    assert_true(sparse.issparse(X_csc))
+    assert_true(sparse.isspmatrix_csc(X_csr.T))  # it changes the sparsity type
+    assert_true(sparse.isspmatrix_csr(X_csc.T))
```

Comment:
I don't know if these checks are necessary. We can as much as rely on scipy to return a sparse_matrix if asked for one :P . Also you haven't used X_csc below.